### ISSUE TITLE
Control plane terminating a Federated Namespace when FederatedNamespacePlacement doesn't have the Host Cluster defined within clusters spec

### ISSUE THREAD

**Issue by mvazquezc** [kubernetes-sigs/federation-v2#396]
Hello,

I just found out some strange behaviour (at least something that I wouldn't expect to happen)

Let's say we have 3 different clusters: cluster1, cluster2, cluster3, the Federation v2 control plane is running on cluster1, so we create all the CRDs related to federation in that cluster.

With the yaml definition below I would expect `foo` namespace being available on cluster2 and cluster3, but what is happening is that the namespace is being terminated on `cluster1` deleting the FederatedNamespacePlacement as well.

I also tried to create the FederatedNamespacePlacement in a different namespace, the behaviour is not the same but the namespace is still not being replicated because (I suppose) the `foo` namespace definition is being deleted.

Do you know if this is the expected behaviour? In that case, How should I create federated namespaces that I don't want to be present on Host Cluster?

Thanks in advanced,

```yaml
apiVersion: v1
kind: List
items:
- apiVersion: v1
  kind: Namespace
  metadata:
    name: foo
- apiVersion: core.federation.k8s.io/v1alpha1
  kind: FederatedNamespacePlacement
  metadata:
    name: foo
    namespace: foo
  spec:
    clusterNames:
    - cluster2
    - cluster3
```

**Comment by marun** [comment]
It is by design that federation requires a namespace to exist in the host cluster if it is to be propagated to any member clusters.  Not including cluster1 in the placement list should not result in the host cluster namespace being removed, however.  I will attempt to reproduce.

**Comment by mvazquezc** [comment]
> @mvazquezc There are some discussion may also help you to understand the current behavior well at [#242 (comment)](https://github.com/kubernetes-sigs/federation-v2/issues/242#issuecomment-418274970)

Thanks for the link. Now I understand the behaviour, thanks!

**Comment by mvazquezc** [comment]
@gyliu513 This type of explanation is great and I believe it would be worth to have it documented somewhere accessible. I've been looking at the docs/ folder and I couldn't find a good document where the differents Federated items where explained. I would find useful a document where we can explain why a namespace should exist in the host cluster in order to federate it to other clusters or such situations. Do you think it is worth it to have something like that?

**Comment by gyliu513** [comment]
@mvazquezc after @marun 's fix at https://github.com/kubernetes-sigs/federation-v2/pull/443 , deleting namespace in host cluster will not cause the deletion of namespace in other member clusters, so seems we do not need document anymore for this?

**Comment by mvazquezc** [comment]
@gyliu513 So even if you delete the namespace on the Host cluster it will remain across the member clusters, that's okay. I've one more question though, what would happen to other federated resources like for example "FederatedNamespacePlacement" or "FederatedDeployment" if they happen to be created in the namespace deleted from the host cluster?

Anyway, I still think a document with expected behaviours would be helpful.

Thank you for your time!

**Comment by gyliu513** [comment]
Seems CI close this automatically, but I think we need tp keep this open as we are still talking if the document need some update.

@shashidharatd can you help open this again? :-)

**Comment by marun** [comment]
- Namespaced resources can't exist without a namespace.
  - This includes namespaced federated resources.  
  - This is as true in the host cluster as in member clusters.  
- Deleting a namespace in the host cluster removes its contents, including federated resources.       
   - Removal of federated resources from the host cluster removes the equivalent target resources from all member clusters
- `FederatedNamespacePlacement` is namespaced, and will be ignored unless it has the same name as the containing namespace.
- `FederatedNamespacePlacement` determines explicit placement for a namespace.
- A namespace without `FederatedNamespacePlacement` will be propagated to all member clusters.
- Excluding a given member cluster from the `FederatedNamespacePlacement` of a namespace will ensure that neither the namespace nor its contents will appear in the excluded member cluster.

I think propagating namespaces by default (i.e. when `FederatedNamespacePlacement` is missing) needs to change.  It's likely that some users will not want to send every namespace in their cluster to member clusters.  A possible next step would be implementing a 'kubefed2 federate namespace` subcommand that adds a `FederatedNamespacePlacement` with a selector for all clusters (requires selector-based placement).

cc: @font

**Comment by gyliu513** [comment]
+1 for this as well, this can make sure the namespace and other resources have same behavior: if no placement policy, the resource will not be propagated; Resources will only be propagated to member clusters with placement policy.

**Comment by shashidharatd** [comment]
I am keen to have a default behaviour to propagate a resource to all member clusters without enumerating the clusterNames in placement object. Something like below:
- If the placement object does not exist for the resource, the resource would not be propogated.
- If the placement object exist but does not list any clusterNames. propogate to all member clusters.
- If the placement object exist and has some clusters listed. propogate to only the clusters listed.

@marun, @font, wdyt?

**Comment by marun** [comment]
@shashidharatd You proposal makes sense for clusterName-based placement.  Assuming selector-based placement, though, I think the default should be 'do not propagate' if a placement resource exists without a value for the selector.

**Comment by marun** [comment]
I think this issue is resolved in master (especially after #545). Feel free to reopen.
